### PR TITLE
Testscript engine update

### PR DESCRIPTION
**Pull request by jawalonoski**
This pull request updates the TestScript engine to execute the examples included with the specification... with the following exceptions

- multiple server tests are not supported
- tests requiring rules or rulesets are not supported

I'll update the engine again in separate branch to handle the connectathon test scripts.

### CODE REVIEW

**Comment by jawalonoski** [comment]
Need to make more updates... done.

File: lib/tests/testscripts/base_testscript.rb [hunk 1]
```diff
@@ -347,33 +359,48 @@ def call_assertion(method, warned, *params)
...
 
-          end if input.include? "${" + var.name + "}"
+            input.gsub!("${#{var.name}}", variable_value)
+          end
         end
 
         if input.include? '${'
-          puts "An unknown variable was unable to be replaced: #{input}!"
-          warning {  assert !input.include?('${'), "An unknown variable was unable to be substituted: #{input}" }
+          unknown_variables = input.scan(/(\$\{)([A-Za-z0-9]+)(\})/).map{|x|x[1]}
```
Comment (okeefm) [inline lib/tests/testscripts/base_testscript.rb]: Are alphanumeric values the only legal values in a variable? e.g. no underscores, etc.? I was looking at the `TestScript` details on the FHIR site, but couldn't find any concrete details one way or the other.

Comment (jawalonoski) [reply lib/tests/testscripts/base_testscript.rb]: I'll add underscores.

**Review by okeefm — APPROVED** [review]
Looks good to me.

**Pull request merged**